protocol/0087-VAMM-automated_market_maker.md

```diff
@@ -0,0 +1,197 @@
+
+We can calculate a scaling factor that is the smaller of a fraction specified in the commitment (`margin_ratio_at_bounds`, either upper or lower depending on the side considered) or the market's worst case margin. If `margin_ratio_at_bounds` for the relevant side is not set then the market's worst case initial margin is taken automatically
+
+$$
+r_f = \min(\frac{1}{m_r}, \frac{1}{ (f_s + f_l) \cdotp f_i}) ,
+$$
+
+where $m_r$ is the sided value `margin_ratio_at_bounds` (`upper ratio` if the upper band is being considered and `lower ratio` if the lower band is), $f_s$ is the market's sided risk factor (different for long and short positions), $f_l$ is the market's linear slippage component and $f_i$ is the market's initial margin factor.
+
+The dollar value at which all margin is utilised will then be
+
+$$
+v_{worst} = c \cdotp r_f
+$$
+
+where $c$ is the commitment amount and $r_f$ is as above.
+
+Calculating this separately for the upper and lower ranges (using the `short` factor for the upper range and the `long` factor for the lower range) one can calculate the liquidity value `L` for each range with the formula
+
+$$
+L = \frac{v_{worst}}{\sqrt{p_u} - \sqrt{p_l}} ,
+$$
+
+where $v_{worst}$ is as above, $p_u$ is the price at the top of the range (`upper price` for the upper range and `base price` for the lower range) and $p_l$ is the price at the bottom of the range (`base price` for the lower range and `lower price` for the lower range). This gives the two `L` values for the two ranges.
+
+#### Fair price
+
+From here the first step is calculating a `fair` price, which can be done by utilising the `L` value for the respective range to calculate `virtual` values for the pool balances. From here on `y` will be the cash balance of the pool and `x` the position.
+
+  1. First, identify the current position, `p`. If it is `0` then the current fair price is the base price.
+  2. If `P > 0`:
+     1. The virtual `x` of the position can be calculated as $x_v = P + \frac{L}{\sqrt{p_l}}$, where $L$ is the value for the lower range, $P$ is the market position and $p_l$ is the `base price`.
+     2. The virtual `y` of the position can be calculated as $y_v = c_c \cdotp r_f + L \cdotp \sqrt{p_l}$ where $c_c$ is the current total dollar balance of the AMM across margin and general accounts and `p_l` is the `lower price`. Other variables are as defined above.
+  3. If `P < 0`:
+     1. The virtual `x` of the position can be calculated as $x_v = P + \frac{c}{p_u} \cdotp r_f + \frac{L}{\sqrt{p_l}}$ where $p_l$ is the `base price` and `p_u` is the `upper price`.
```

Comment:
Just confirming whether this `c` here is supposed to be `c_c` as above (current total balance), or if it is actually referring to the `c` for the initial committment all the way at the top of the file?